packages/shared/src/general.ts

```diff
@@ -50,6 +50,7 @@ export const isFunction = (val: unknown): val is Function =>
   typeof val === 'function'
 export const isString = (val: unknown): val is string => typeof val === 'string'
 export const isSymbol = (val: unknown): val is symbol => typeof val === 'symbol'
+export const isUndefined = (val: unknown): val is undefined => val === undefined
```

Comment:
should be `typeof val === 'undefined'`

Reply:
How is that different (except it's longer)?

Reply:
![image](https://github.com/user-attachments/assets/d2f684e7-c8b2-432d-abeb-8773f181bfce)

Reply:
This is difference when the variable is or isn't defined, but in case of the function argument it's always defined so it doesn't apply here

Reply:
But passing an undefined variable to a function, per spec (and as implemented by all runtimes), is a `ReferenceError`.

Arguments are evaluated in step 3, before running function code (in step 7):
https://tc39.es/ecma262/#sec-evaluatecall